src/game/shell.c

```diff
@@ -41,12 +41,12 @@ static char *Shell_GetScreenshotName(void)
 {
     // Get level title of unknown length
     char level_title[100];
-    snprintf(
+    size_t out_size = snprintf(
         level_title, LEVEL_TITLE_SIZE, "%s",
         g_GameFlow.levels[g_CurrentLevel].level_title);
 
     // Trim level titles that are too long
-    if (strlen(level_title) >= LEVEL_TITLE_SIZE) {
+    if (out_size >= LEVEL_TITLE_SIZE) {
```

Comment:
'If the resulting string would be longer than n-1 characters, the remaining characters are discarded and not stored, but counted for the value returned by the function.

A terminating null character is automatically appended after the content written.'
So I don't think we need this check or trim at all? snprintf should already do that.

Reply:
It depends.
In the past, on Windows, `snprintf()` was historically made available as a macro to `_snprintf()`, which doesn't add a terminating null character if the size exceedes. Recent updates have a compliant `snprintf()` implementation, but older does not.

Reply:
Agreed. Which is why I think it's fine to support `snprintf` as defined [here](https://cplusplus.com/reference/cstdio/snprintf/).

Reply:
Perhaps, another alternative could be also to use `SDL_snprintf()`, since the SDL has a dedicated function if the system library does not provide an implementation compliant to the standard.

https://github.com/libsdl-org/SDL/blob/9b861d2ea4fbae0718a0c1ac37397ed5ba2a5b67/src/stdlib/SDL_string.c#L1429

Reply:
That's an option but personally I'm against coupling with SDL more than necessary.

Reply:
At this point, after removing the test on the final NUL character, `out_size` can be removed from the first `snprintf()` and restored at its previous place. In my opinion, `snprintf()` is not expected to return `-1` here since there are no file writes and the format parameter does not include encoding conversions.

Reply:
> That's an option but personally I'm against coupling with SDL more than necessary.

Or it could be possible to do this, as it has been done into `filesystem.c` and `text.c`:

```
strncpy(level_title, g_GameFlow.levels[g_CurrentLevel].level_title, LEVEL_TITLE_SIZE-1);
level_title[LEVEL_TITLE_SIZE] = '\0';
```

which solves the trouble about the possibility of returning an error code at the root, but not issue #445 yet.

Reply:
`strcpy` looks most straightforward TBH.
BTW I think this discussion grew unproportionally big in regards to the impact of this change.